packages/mitex/specs/prelude.typ

```diff
@@ -105,6 +105,33 @@
   ), if handle != none { (alias: alias, handle: handle) } else { none })
 }
 
+/// Define an math environment with a fixed number of arguments, like \begin{alignedat}{2}
+/// 
+/// Arguments:
+/// - num (int): The number of arguments as environment options for the environment.
+/// - alias (str): Alias command for typst handler.
+///   For example, alias `\begin{alignedat}{2}` to typst's `alignedat`,
+///   and alias `\begin{aligned}` to typst's `aligned`, as the key in mitex-scope.
+/// - handle (function): The handler function, as the value of alias in mitex-scope.
+///   It receives fixed number of named arguments as environment options,
+///   for example `alignedat(arg0: ..)` or `alignedat(arg0: .., arg1: ..)`.
+///   And it receives variable length arguments as environment body,
+///   Therefore you need to use `(.. arg) = > {..}` to receive them.
+///
+/// Return: A spec item and a scope item (none for no scope item)
+#let define-math-env(num, alias: none, handle: none) = {
```

Comment:
we can have a common `define-env` and the rest of `define-xxx-env` are just alias to `define-env.with(ctx_feature: ...)`